cmd/buildtools/k0s.go

```diff
@@ -13,30 +13,56 @@ import (
 	"github.com/urfave/cli/v2"
 )
 
-var k0sComponents = []struct {
-	name        string
-	makefileVar string
-}{
-	{
-		name:        "coredns",
+var k0sImageComponents = map[string]string{
+	"quay.io/k0sproject/coredns":                    "coredns",
+	"quay.io/k0sproject/calico-node":                "calico-node",
+	"quay.io/k0sproject/calico-cni":                 "calico-cni",
+	"quay.io/k0sproject/calico-kube-controllers":    "calico-kube-controllers",
+	"registry.k8s.io/metrics-server/metrics-server": "metrics-server",
+	"quay.io/k0sproject/kube-proxy":                 "kube-proxy",
+}
+
+var k0sComponents = map[string]addonComponent{
+	"coredns": {
+		getWolfiPackageName: func(k0sVersion *semver.Version, upstreamVersion string) string {
+			return "coredns"
+		},
 		makefileVar: "COREDNS_VERSION",
 	},
-	{
-		name:        "calico-node",
+	"calico-node": {
+		getWolfiPackageName: func(k0sVersion *semver.Version, upstreamVersion string) string {
+			return "calico-node"
+		},
 		makefileVar: "CALICO_NODE_VERSION",
 	},
-	{
-		name:        "calico-cni",
+	"calico-cni": {
+		getWolfiPackageName: func(k0sVersion *semver.Version, upstreamVersion string) string {
+			return "calico-cni"
+		},
 		makefileVar: "CALICO_CNI_VERSION",
 	},
-	{
-		name:        "calico-kube-controllers",
+	"calico-kube-controllers": {
+		getWolfiPackageName: func(k0sVersion *semver.Version, upstreamVersion string) string {
+			return "calico-kube-controllers"
+		},
 		makefileVar: "CALICO_KUBE_CONTROLLERS_VERSION",
 	},
-	{
-		name:        "metrics-server",
+	"metrics-server": {
+		getWolfiPackageName: func(k0sVersion *semver.Version, upstreamVersion string) string {
+			return "metrics-server"
+		},
 		makefileVar: "METRICS_SERVER_VERSION",
 	},
+	"kube-proxy": {
+		getWolfiPackageName: func(k0sVersion *semver.Version, upstreamVersion string) string {
+			return fmt.Sprintf("kube-proxy-%d.%d-default", k0sVersion.Major(), k0sVersion.Minor())
+		},
+		getWolfiPackageVersionComparison: func(k0sVersion *semver.Version, upstreamVersion string) string {
+			// match the greatest patch version of the same minor version
+			return fmt.Sprintf(">=%d.%d, <%d.%d", k0sVersion.Major(), k0sVersion.Minor(), k0sVersion.Major(), k0sVersion.Minor()+1)
+		},
+		makefileVar: "KUBE_PROXY_VERSION",
```

Comment:
eventuallly we should move these to a metadata.yaml file as ricardo has for the addons

Reply:
i'll do that on the next pr